### PR TITLE
fix: add dummy user and adjust deletion of shared posts

### PR DESCRIPTION
**Pull request by rebelchris**
Introduces dummy user that gets set for deleted users shared posts

Also perhaps we should consider adding this for welcome/freeform to?

WT-1317

### CODE REVIEW

File: src/migration/1683721525235-DummyUser.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class DummyUser1683721525235 implements MigrationInterface {
+  name = 'DummyUser1683721525235';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `INSERT INTO "public"."user" ("id", "name", "image", "username") VALUES ('404', 'Inactive user', 'https://daily-now-res.cloudinary.com/image/upload/f_auto/placeholders/placeholder3', 'inactive_user')`,
```
Comment (idoshamun) [inline src/migration/1683721525235-DummyUser.ts:8]:
```suggestion
      `INSERT INTO "public"."user" ("id", "name", "image", "username") VALUES ('404', 'Inactive user', 'https://daily-now-res.cloudinary.com/image/upload/f_auto,q_auto/placeholders/placeholder3', 'inactive_user')`,
```

File: src/directive/user.ts [hunk 2]
```diff
@@ -41,6 +42,10 @@ export const deleteUser = async (
       await entityManager.getRepository(SourceDisplay).delete({ userId });
       await entityManager.getRepository(SourceRequest).delete({ userId });
       await entityManager.getRepository(Upvote).delete({ userId });
+      // Manually set shared post to 404 dummy user
+      await entityManager
+        .getRepository(SharePost)
+        .update({ authorId: userId }, { authorId: '404' });
       await entityManager
```
Comment (idoshamun) [inline src/directive/user.ts:49]: This will override the above

File: src/directive/user.ts [hunk 3]
```diff
@@ -41,6 +42,10 @@ export const deleteUser = async (
       await entityManager.getRepository(SourceDisplay).delete({ userId });
       await entityManager.getRepository(SourceRequest).delete({ userId });
       await entityManager.getRepository(Upvote).delete({ userId });
+      // Manually set shared post to 404 dummy user
+      await entityManager
```
Comment (idoshamun) [inline src/directive/user.ts:46]: I think we should apply it on every post that is no `ArticlePost`.
Maybe we should apply it also for comments?

Comment (rebelchris) [reply src/directive/user.ts:46]: Good point, can do that actually :) 
And also for comments seems like a good way to preserve conversations.

**Pull request merged**